quickjs.c

```diff
@@ -6668,7 +6669,20 @@ static void build_backtrace(JSContext *ctx, JSValue error_obj,
     if (filename && (backtrace_flags & JS_BACKTRACE_FLAG_SINGLE_LEVEL))
         goto done;
 
-    for (sf = rt->current_stack_frame; sf != NULL && i < stack_trace_limit; sf = sf->prev_frame) {
+    sf_start = rt->current_stack_frame;
+
+    /* Find the frame we want to start from. NOte that when a filter is used the filterd
+       function will be the first, but we also specify we want to skip the fist one. */
+    if (backtrace_flags & JS_BACKTRACE_FLAG_FILTER_FUNC) {
+        for (sf = sf_start; sf != NULL && i < stack_trace_limit; sf = sf->prev_frame) {
+            if (JS_IsSameValue(ctx, sf->cur_func, filter_func)) {
```

Comment:
```suggestion
            if (js_same_value(ctx, sf->cur_func, filter_func)) {
```
Not that it's wrong but internally we overwhelmingly call js_same_value. There's also (maybe) a minor performance benefit in that the compiler can assume it's not interposed.